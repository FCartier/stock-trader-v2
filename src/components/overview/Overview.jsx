import React from "react";
import { SectionHeader } from "../reusables/SectionHeader"

const Overview = ({overview}) => {
  return Object.keys(overview).length ? (
    <div>
        <div className="overview">
          <SectionHeader title="COMPANY OVERVIEW"/>
          <div className="overview-content">
            <h2>{overview.companyName}</h2>
            <p>{overview.website}</p>
            <p>{overview.description}</p>
          </div>
        </div>
    </div>
  ) : null
};

export default Overview;
