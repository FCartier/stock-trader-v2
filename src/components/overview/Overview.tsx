import * as React from "react";
import { SectionHeader } from "../reusables/SectionHeader"

export interface IOverviewProps {
  overview?: {
      companyName: string;
      website: string;
      description: string;
  };
}

const Overview: React.SFC<IOverviewProps> = ({overview}) => {
  return !(typeof overview === "undefined" || overview.companyName === "") ? (
    <div>
      <div className='overview'>
        <SectionHeader title="COMPANY OVERVIEW" />
        <div className='overview-content'>
          <h2>{overview.companyName}</h2>
          <p>{overview.website}</p>
          <p>{overview.description}</p>
        </div>
      </div>
    </div>
  ) : null;
};

export default Overview;
