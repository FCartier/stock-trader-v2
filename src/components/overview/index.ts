import OverviewContainer from "./overviewContainer";

export default OverviewContainer;