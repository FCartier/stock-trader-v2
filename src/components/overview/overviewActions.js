export const OVERVIEW_ACTION = "OVERVIEW ACTION";

export const overviewAction = payload => ({
    type: OVERVIEW_ACTION,
    payload
  });