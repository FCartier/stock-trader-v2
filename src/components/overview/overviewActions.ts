export const FETCH_OVERVIEW_SUCCESS = "FETCH_OVERVIEW_SUCCESS";
export type FETCH_OVERVIEW_SUCCESS = typeof FETCH_OVERVIEW_SUCCESS;

export const FETCH_OVERVIEW_FAILED = "FETCH_OVERVIEW_FAILED";
export type FETCH_OVERVIEW_FAILED = typeof FETCH_OVERVIEW_FAILED;