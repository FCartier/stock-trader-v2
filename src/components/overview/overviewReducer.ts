import { Reducer } from "redux";
import {
  FETCH_OVERVIEW_SUCCESS,
  FETCH_OVERVIEW_FAILED
} from "./overviewActions";
import OverviewState from "./types";
import { fetchStatus } from "../reusables/fetchStatus";

const initialState: OverviewState = {
  overview: {
    companyName: "",
    website: "",
    description: ""
  },
  status: fetchStatus.pending
};

export const overview: Reducer<OverviewState> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case FETCH_OVERVIEW_SUCCESS:
      return { overview: action.payload, status: fetchStatus.success };
    case FETCH_OVERVIEW_FAILED:
      return { ...state, status: fetchStatus.failed };
    default:
      return state;
  }
};
