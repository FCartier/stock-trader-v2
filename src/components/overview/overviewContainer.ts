import Overview from './Overview'
import { connect } from 'react-redux'
import { GlobalState } from '../../StoreTypes'

const mapStateToProps = (state: GlobalState) => ({
  overview: state.overview
})

export default connect(
    mapStateToProps,
    null
  )(Overview)