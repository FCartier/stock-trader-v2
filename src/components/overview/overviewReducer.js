import { OVERVIEW_ACTION } from "./overviewActions"

export const overview = (state = {}, action) => {
    switch (action.type) {
      case OVERVIEW_ACTION:
        return action.payload;
      default:
        return state;
    }
  };