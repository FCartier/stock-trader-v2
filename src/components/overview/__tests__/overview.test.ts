import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import Overview from "../Overview";
import { testData } from "../__data__/overview.data";
import { fetchStatus } from "../../reusables/fetchStatus";

configure({ adapter: new Adapter() });

describe("overview component", () => {
  test("overview component should display all data fields", () => {
    const component = shallow(
      Overview({ overview: testData, status: fetchStatus.success })
    );
    expect(component).toMatchSnapshot();
  });
});
