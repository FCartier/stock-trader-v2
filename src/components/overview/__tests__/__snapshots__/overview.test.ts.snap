// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview component overview component should display all data fields 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <div
      className="overview"
    >
      <Unknown
        title="COMPANY OVERVIEW"
      />
      <div
        className="overview-content"
      >
        <h2>
          American Assets Trust Inc.
        </h2>
        <p>
          http://www.americanassetstrust.com
        </p>
        <p>
          American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
        </p>
      </div>
    </div>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="overview"
      >
        <Unknown
          title="COMPANY OVERVIEW"
        />
        <div
          className="overview-content"
        >
          <h2>
            American Assets Trust Inc.
          </h2>
          <p>
            http://www.americanassetstrust.com
          </p>
          <p>
            American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            title="COMPANY OVERVIEW"
          />,
          <div
            className="overview-content"
          >
            <h2>
              American Assets Trust Inc.
            </h2>
            <p>
              http://www.americanassetstrust.com
            </p>
            <p>
              American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
            </p>
          </div>,
        ],
        "className": "overview",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "COMPANY OVERVIEW",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                American Assets Trust Inc.
              </h2>,
              <p>
                http://www.americanassetstrust.com
              </p>,
              <p>
                American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
              </p>,
            ],
            "className": "overview-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "American Assets Trust Inc.",
              },
              "ref": null,
              "rendered": "American Assets Trust Inc.",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "http://www.americanassetstrust.com",
              },
              "ref": null,
              "rendered": "http://www.americanassetstrust.com",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.",
              },
              "ref": null,
              "rendered": "American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="overview"
        >
          <Unknown
            title="COMPANY OVERVIEW"
          />
          <div
            className="overview-content"
          >
            <h2>
              American Assets Trust Inc.
            </h2>
            <p>
              http://www.americanassetstrust.com
            </p>
            <p>
              American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              title="COMPANY OVERVIEW"
            />,
            <div
              className="overview-content"
            >
              <h2>
                American Assets Trust Inc.
              </h2>
              <p>
                http://www.americanassetstrust.com
              </p>
              <p>
                American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
              </p>
            </div>,
          ],
          "className": "overview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "COMPANY OVERVIEW",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  American Assets Trust Inc.
                </h2>,
                <p>
                  http://www.americanassetstrust.com
                </p>,
                <p>
                  American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.
                </p>,
              ],
              "className": "overview-content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "American Assets Trust Inc.",
                },
                "ref": null,
                "rendered": "American Assets Trust Inc.",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "http://www.americanassetstrust.com",
                },
                "ref": null,
                "rendered": "http://www.americanassetstrust.com",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.",
                },
                "ref": null,
                "rendered": "American Assets Trust Inc is a self-administered real estate investment trust based in the United States. The company mainly invests in, operates, and develops retail, office, residential, and mixed-use properties in California, Oregon, and Hawaii.",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
